lhotse/dataset/speech_recognition.py

```diff
@@ -57,3 +58,122 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+
+    .. code-block::
+
+        {
+            'features': (T x F) tensor,
+            'supervisions': List[Dict] -> [
+                {
+                    'text': string,
+                    'fsa': k2.Fsa,
+                    'start_frame': int,
+                    'end_frame': int
+                } (multiplied N times, for each of the N supervisions present in the Cut)
+            ],
+            'supervisions_mask': (T) tensor
+        }
+
+    The ``supervisions_mask`` field is a mask that specifies which frames are covered by a supervision
+    by assigning a value of 1 (in this case: segments with transcribed speech contents),
+    and which are not by asigning a value of 0 (in this case: padding, contextual noise,
+    or in general the acoustic context without transcription).
+    """
+
+    def __init__(self, cuts: CutSet):
+        import k2
+        super().__init__()
+        self.cuts = cuts
+        self.cut_ids = list(self.cuts.ids)
+
+        # We're creating a vocabulary of all characters present in the training CutSet.
+        # To do that we iterate through all the supervisions, and use chain.from_iterable
+        # to convert a sequence of their 'text' fields into a flat iterable of characters.
+
+        # Add special symbols first
+        self.vocab = ['<eps>', '<sil>', '<unk>'] + sorted(
+            # Add symbols present in the corpus
+            set(
+                chain.from_iterable(
+                    list(supervision.text)
+                    for cut in self.cuts
+                    for supervision in cut.supervisions
+                )
+            )
+        )
+        self.symbol_table = k2.SymbolTable(
+            _id2sym={i: tok for i, tok in enumerate(self.vocab)},
+            _sym2id={tok: i for i, tok in enumerate(self.vocab)}
+        )
+
+    def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+        import k2
+        cut_id = self.cut_ids[idx]
+        cut = self.cuts[cut_id]
+
+        features = torch.from_numpy(cut.load_features())
+        mask = torch.from_numpy(cut.supervisions_feature_mask())
+
+        sil_id = self.symbol_table.get('<sil>')
+        eps_id = self.symbol_table.get('<eps>')
+
+        supervision_fsas = []
+        for supervision in cut.supervisions:
+            # TODO(pzelasko): I am not sure how to attach the symbol table to the FSA object
+            #                 (or whether that's need at all?)
+
+            # Creating the core supervision FSA corresponding to the supervision text.
+            fsa = k2.linear_fsa([self.symbol_table.get(c) for c in supervision.text])
+
+            # TODO: Adding optional silence at beginning and end.
+
+            # TODO(pzelasko): Should be possible to create it with sth like:
+            #   silfsa = k2.union(
+            #     k2.linear_fsa([sil_id]), k2.linear_fsa([eps_id])
+            #   ).closure()
+            silfsa = k2.Fsa.from_str(f'''0 1 {sil_id} 0
```

Comment:
This string is in the OpenFST format. Please use `k2.Fsa.from_openfst`.

`k2.Fsa.from_str` accepts strings only in k2 format.

Reply:
Is K2 format documented somewhere? I thought it's `<from-state> <to-state> <label> <weight>`

Reply:
The documentation is at: https://github.com/k2-fsa/k2/blob/3363292e3b2d4554eee025d2950a4c4ac20b6c01/k2/python/k2/fsa.py#L147

CAUTION: Arcs entering the final state have label `-1`.

```
       The given string `s` consists of lines with the following format:
        (1) When it represents an acceptor:
                src_state dest_state label score
        (2) When it represents a transducer:
                src_state dest_state label aux_label score
        The line for the final state consists of only one field:
                final_state
        Note:
          Fields are separated by space(s), tab(s) or both. The `score`
          field is a float, while other fields are integers.
        Caution:
          The first column has to be non-decreasing.
        Caution:
          The final state has the largest state number. There is only
          one final state. All arcs that are connected to the final state
          have label -1.
```